### PR TITLE
Fixes #25654 - use generator-react-domain for creating React/Redux components

### PR DESCRIPTION
**Pull request by glekner**
I created a generator that can help us (After we agreed that the shell script is not the right solution)

[generator-react-domain
](https://github.com/glekner/generator-react-domain)

Any feedback is appreciated.

### CODE REVIEW

File: webpack/stories/docs/addingNewComponent.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Adding components
 
+ ## Component Generator
+We are using `generator-react-domain` to help you generate your component easily.
+```sh
+npm -i -g yo generator-react-domain       # install the generator globally
```
Comment (ohadlevy) [inline webpack/stories/docs/addingNewComponent.md]: is this required? (this is just to avoid adding it as a dev dep?)

Comment (ohadlevy) [reply webpack/stories/docs/addingNewComponent.md]: should be npm i (without the -)

**Review by ohadlevy — CHANGES_REQUESTED** [review]
this looks great, i have a few comments:

after installing, the following is changed on my branch, perhaps this should be part of the PR:
```diff
index f67f7ba9e..b6871433a 100644
--- a/.yo-rc.json
+++ b/.yo-rc.json
@@ -1,5 +1,6 @@
 {
   "generator-react-domain": {
-    "componentsPath": "webpack/assets/javascripts/react_app/components"
+    "componentsPath": "webpack/assets/javascripts/react_app/components",
+    "depsInstalled": true
   }
-}
+}
\ No newline at end of file
diff --git a/package.json b/package.json
index 22306408a..7b30d726c 100644
--- a/package.json
+++ b/package.json
@@ -46,6 +46,7 @@
     "prettier": "^1.13.5",
     "raf": "^3.4.0",
     "raw-loader": "^0.5.1",
+    "react-redux-test-utils": "^0.1.1",
     "react-remarkable": "^1.1.3",
     "react-test-renderer": "^16.2.0",
     "redux-mock-store": "^1.2.2",
```

secondly, we create an empty helper, fixtures,css files, are those required? perhaps we either skip it, or add something to it? (e.g. some basic docs?)

when trying the redux option, i got an error:
```
npm run create-react-component -- --redux                                                                                                                                    git:(6353|✚2 

> TheForemanDevDeps@1.21.0 create-react-component /home/ohad/git/foreman
> yo react-domain "--redux"


react-domain-generator

? Enter your Component name awesome
events.js:167
      throw er; // Unhandled 'error' event
      ^

AssertionError [ERR_ASSERTION]: Trying to copy from a source that does not exist: /home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/lib/generators/gen/redux/templates/actions.js
    at EditionInterface.exports.copy (/home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/node_modules/mem-fs-editor/lib/actions/copy.js:48:3)
    at EditionInterface.module.exports [as copyTpl] (/home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/node_modules/mem-fs-editor/lib/actions/copy-tpl.js:29:8)
    at templates.forEach.file (/home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/lib/generators/gen/base/index.js:26:39)
    at Array.forEach (<anonymous>)
    at ReduxGenerator.copyTemplates (/home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/lib/generators/gen/base/index.js:26:15)
    at ReduxGenerator.writing (/home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/lib/generators/gen/redux/index.js:13:10)
    at Object.<anonymous> (/home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/node_modules/yeoman-generator/lib/index.js:399:25)
    at /home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/node_modules/run-async/index.js:25:25
    at new Promise (<anonymous>)
    at /home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/node_modules/run-async/index.js:24:19
Emitted 'error' event at:
    at Immediate.setImmediate (/home/ohad/.nvm/versions/node/v10.5.0/lib/node_modules/generator-react-domain/node_modules/yeoman-generator/lib/index.js:406:18)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
    at processImmediate (timers.js:649:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! TheForemanDevDeps@1.21.0 create-react-component: `yo react-domain "--redux"`
```

**Comment by glekner** [comment]
I think helper, fixtures,css files should remain empty for other consumers.
We can create our own templates folder with those files. let me know what you think

Can you try v0.1.15 with `--redux` now? @ohadlevy

**Comment by glekner** [comment]
@ohadlevy its still broken, ill update when fixed

**Comment by glekner** [comment]
> after installing, the following is changed on my branch, perhaps this should be part of the PR

the `.yo-rc.json` gets filled after the generator runs for the first time, it checks if dependencies (here `react-redux-test-utils`) were installed so we shouldn't fill it in the PR.

**Comment by ohadlevy** [comment]
> > after installing, the following is changed on my branch, perhaps this should be part of the PR
> 
> the `.yo-rc.json` gets filled after the generator runs for the first time, it checks if dependencies (here `react-redux-test-utils`) were installed so we shouldn't fill it in the PR.

and then what? should a contributor recommit those changes? instead of everyone going though this (e.g. config files should either be in gitignore or committed) shouldn't we just include it in the PR? also, what does react-redux-test-utils ? can it be a dependency of the other package instead? (so its always there but not required to change foreman for it?)

**Comment by glekner** [comment]
@ohadlevy My bad! `.yo-rc.json` should be in gitignore.

since its recommended to install the generator globally we can't put `react-redux-test-utils` in its dependencies.

@sharvit created `react-redux-test-utils`,  basically creating a repo for [testHelper.js](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/common/testHelpers.js).

Should we just add it to our package.json and delete our own testHelper? Thoughts @sharvit ?

**Comment by glekner** [comment]
added `react-redux-test-utils` to package.json.

Will delete our own testHelper in a future PR.

File: package.json [hunk 2]
```diff
@@ -43,6 +43,7 @@
     "prettier": "^1.13.5",
     "raf": "^3.4.0",
     "raw-loader": "^0.5.1",
+    "react-redux-test-utils": "^0.1.1",
```
Comment (ohadlevy) [inline package.json:46]: this should not be needed for packaging, hence does not require a packaging PR. /cc @ekohl

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**